Source/DafnyDriver/Commands/FilterOption.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Boogie;
+
+namespace Microsoft.Dafny; 
+
+public class FilterOption : CommandLineOption<IEnumerable<string>> {
+  public static readonly FilterOption Instance = new();
+
+  public override object DefaultValue => Enumerable.Empty<string>();
+  public override string LongName => "filter";
+  public override string ArgumentName => "pattern";
+
+  public override string Description => @"
+(experimental) Only check proofs matched by pattern <p>. This option may be specified multiple times to match multiple patterns. The pattern <p> may contain * wildcards which match any character zero or more times.".TrimStart();
```

Comment:
This is a bit unclear.  Even if its unsatisfactory, we might want to say explicitly that this matches Boogie names?

Reply:
Why would we want to mention Boogie? The user should be able to filter in terms of their Dafny proofs.

Reply:
I was suggesting a documentation improvement: what I meant was that the option currently takes Boogie names (doesn't it?), and that while this isn't ideal, at least we should document it so that users know what to pass (it's not clear to me what it means that a proof is matched by a pattern).

Reply:
I would actually prefer we name this `--boogie-filter`, to be clearer about its semantics and also to let us define `--filter` later on, if and when it's reasonable to implement in terms of Dafny concepts. Then we can document the latter as filtering on names of assertion batches.